src/app/loadout-builder/loadout-builder-reducer.ts

```diff
@@ -525,17 +544,66 @@ export function useLbState(
 ) {
   const savedLoadoutBuilderParameters = useSelector(savedLoadoutParametersSelector);
   const savedStatConstraintsPerClass = useSelector(savedLoStatConstraintsByClassSelector);
-  return useReducer(
-    lbStateReducer(defs),
-    {
+
+  const {
+    state: lbConfState,
+    setState,
+    redo,
+    undo,
+    canRedo,
+    canUndo,
+  } = useHistory(
+    lbConfigInit({
       stores,
       defs,
       preloadedLoadout,
       initialClassType,
       initialLoadoutParameters,
       savedLoadoutBuilderParameters,
       savedStatConstraintsPerClass,
+    })
+  );
+
+  const lbConfReducer = useMemo(() => lbConfigReducer(defs), [defs]);
+
+  const [lbUIState, lbUIDispatch] = useReducer(lbUIReducer, {
+    compareSet: undefined,
+    modPicker: { open: false },
+  });
+
+  const dispatch = useCallback(
+    (action: LoadoutBuilderAction) => {
+      switch (action.type) {
+        case 'undo':
+          undo();
+          lbUIDispatch({ type: 'closeCompareDrawer' });
+          lbUIDispatch({ type: 'closeModPicker' });
+          break;
+        case 'redo':
+          redo();
+          lbUIDispatch({ type: 'closeCompareDrawer' });
+          lbUIDispatch({ type: 'closeModPicker' });
+          break;
+        case 'openCompareDrawer':
+        case 'closeCompareDrawer':
+        case 'openModPicker':
+        case 'closeModPicker':
+          lbUIDispatch(action);
+          break;
+        default:
+          setState((oldState) => lbConfReducer(oldState, action));
+          break;
+      }
     },
-    lbStateInit
+    [lbConfReducer, redo, setState, undo]
   );
+
+  return [
+    {
+      ...lbConfState,
+      ...lbUIState,
+    },
+    dispatch,
+    { canUndo, canRedo },
```

Comment:
Any reason to break these out like this instead of returning them in the big state blob?